lib/glific/flows/flow_count.ex

```diff
@@ -70,27 +70,43 @@ defmodule Glific.Flows.FlowCount do
     |> Repo.all()
   end
 
+  @doc """
+  Create flow count
+  """
+  @spec create_flow_count(map()) :: {:ok, FlowCount.t()} | {:error, Ecto.Changeset.t()}
+  def create_flow_count(attrs) do
+    %FlowCount{}
+    |> FlowCount.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  Update flow count
+  """
+  @spec update_flow_count(FlowCount.t(), map()) ::
+          {:ok, FlowCount.t()} | {:error, Ecto.Changeset.t()}
+  def update_flow_count(%FlowCount{} = flow_count, attrs) do
+    flow_count
+    |> FlowCount.changeset(attrs)
+    |> Repo.update()
+  end
+
   @doc """
   Upsert flow count
   """
   @spec upsert_flow_count(map()) :: :error | FlowCount.t()
   def upsert_flow_count(%{flow_uuid: nil} = _attrs), do: :error
 
-  def upsert_flow_count(%{flow_uuid: flow_uuid} = attrs) do
-    case Repo.fetch_by(Flow, %{uuid: flow_uuid}) do
-      {:ok, flow} ->
-        attrs = Map.merge(attrs, %{flow_id: flow.id})
-
-        Repo.insert!(
-          FlowCount.changeset(%FlowCount{}, attrs),
-          on_conflict: [inc: [count: 1]],
-          returning: true,
-          conflict_target: [:flow_id, :uuid, :type]
-        )
-        |> update_recent_messages(attrs)
-
-      {status, flow} ->
-        {status, flow}
+  def upsert_flow_count(attrs) do
+    with {:ok, flowcount} <- Repo.fetch_by(FlowCount, %{uuid: attrs.uuid}),
+         {:ok, flowcount} <-
+           update_flow_count(flowcount, Map.merge(attrs, %{count: flowcount.count + 1})) do
+      update_recent_messages(flowcount, attrs)
+    else
+      _ ->
+        with {:ok, flowcount} <- create_flow_count(attrs) do
+          update_recent_messages(flowcount, attrs)
+        end
```

Comment:
How about this:

- Always use update_all (works most of the time) and should update only 1 entry
- If update_all does not update any entries (i.e does not exist) do a create

that way we basically do 1 query only on an average, rather than 2

update_recent_messages, does another query, which we should avoid and incorporate into this query also

ideally we want to minimize the number of queries, since this is in an inner loop and called often